tests/fix_proxy.py

```diff
@@ -110,6 +101,17 @@ def stop(self):
         logging.info("proxy stopped")
         self.proc = None
 
+    @contextmanager
+    def deaf_listen(self):
```

Comment:
The method name is not quite self-explanatory to me; would you mind adding a tiny docstring explaining what this does functionally? (Something like what was on the previous `deaf_port()`.)

Reply:
Sure, will do, thank you. It will do something like:

> Open the proxy port to listen, but without accepting a connection.
>
> A program trying to connect will block. Useful to test connection timeouts.

Will commit and merge later, today I'm not in front of a computer